cookbooks/calico/recipes/general_compute.rb

```diff
@@ -327,6 +332,7 @@
     })
     owner "bird"
     group "bird"
+    not_if { get_ipv6.call(node).empty? }
```

Comment:
i think you are missing a similar check in the template to not add a section of the bird template if a particular node is not IPv6 enabled.